---
Author: Valisha Shah
title: "R Notebook"
output: github_document
---

```{r}

library(maftools)

```

```{r}
clinical.data = system.file('extdata', 'tcga_laml_annot.tsv', package = 'maftools') 
#na.omit(clinical.data)
laml.f <- read.maf(maf = "tcga_laml.maf.gz", clinicalData = clinical.data) #create a MAF object
laml.df = subsetMaf(laml.f, mafObj=FALSE) #creates a data.frame of laml.maf - this can be viewed as a table 
head(laml.df)
head(clinical.data)
```


```{r}
laml.df
getSampleSummary(laml.f)
getGeneSummary(laml.f)
getClinicalData(laml.f)
getFields(laml.f)
write.mafSummary(maf = laml.f, basename = 'laml')
```
Plotting MAF summary
```{r}

plotmafSummary(maf = laml.f, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```

drawing oncoplots 
```{r}
oncoplot(maf = laml.f, top = 10)
oncostrip(maf = laml.f, genes = c('DNMT3A','NPM1', 'RUNX1'))


```
Transitions and Trasnversions
```{r}
laml.titv = titv(maf = laml.f, plot = FALSE, useSyn = TRUE)
plotTiTv(res = laml.titv)
```
lollipop plots for amino acid changes
```{r}
lollipopPlot(maf = laml.f, gene = 'DNMT3A', AACol = 'Protein_Change', showMutationRate = TRUE)
lollipopPlot(maf = laml.f, gene = "KIT", AACol = "Protein_Change", labelPos = 816, refSeqID = "NM_000222")

```
Rainfall plots 
```{r}
brca <- system.file("extdata", "brca.maf.gz", package = "maftools")
brca = read.maf(maf = brca, verbose = FALSE)
rainfallPlot(maf = brca, detectChangePoints = TRUE, pointSize = 0.6)

```
Compare mutation load against TCGA cohorts 
```{r}
laml.mutload = tcgaCompare(maf = laml.f, cohortName = 'Example-LAML')
```
Plotting VAF 
```{r}
plotVaf(maf = laml.f, vafCol = 'i_TumorVAF_WU')

```
Genecloud 
```{r}
geneCloud(input = laml.f, minMut = 3)

```
Somatic Interactions 
```{r}
somaticInteractions(maf = laml.f, top = 25, pvalue = c(0.05, 0.1))

```
Detecting cancer driver genes based on positional clustering 
```{r}
laml.sig = oncodrive(maf = laml.f, AACol = 'Protein_Change', minMut = 5, pvalMethod = 'zscore')
head(laml.sig)
plotOncodrive(res = laml.sig, fdrCutOff = 0.1, useFraction = TRUE)
```
Adding and summarizing pfam domains
```{r}
laml.pfam = pfamDomains(maf = laml.f, AACol = 'Protein_Change', top = 10)
laml.pfam$proteinSummary[,1:7, with = FALSE]
#Domain summary (Printing first 3 columns for display convenience)
laml.pfam$domainSummary[,1:3, with = FALSE]
```
Pan-Cancer Comparison
```{r}
#MutsigCV results for TCGA-AML
laml.mutsig <- system.file("extdata", "LAML_sig_genes.txt.gz", package = "maftools")
pancanComparison(mutsigResults = laml.mutsig, qval = 0.1, cohortName = 'LAML', inputSampleSize = 200, label = 1)

```
Survival Analysis
```{r}

# mutation in any given genes
#Survival analysis based on grouping of DNMT3A mutation status
mafSurvival(maf = laml.f, genes = 'DNMT3A', time = 'days_to_last_followup', Status = 'Overall_Survival_Status', isTCGA = TRUE)
```
Predict genesets associated with survival 
```{r}

#Using top 20 mutated genes to identify a set of genes (of size 2) to predict poor prognostic groups
prog_geneset = survGroup(maf = laml.f, top = 20, geneSetSize = 2, time = "days_to_last_followup", Status = "Overall_Survival_Status", verbose = FALSE)
print(prog_geneset)
mafSurvGroup(maf = laml.f, geneSet = c("DNMT3A", "FLT3"), time = "days_to_last_followup", Status = "Overall_Survival_Status")
```
Comparing two cohorts 
```{r}
#Primary APL MAF
primary.apl = system.file("extdata", "APL_primary.maf.gz", package = "maftools")
primary.apl = read.maf(maf = primary.apl)
#Relapse APL MAF
relapse.apl = system.file("extdata", "APL_relapse.maf.gz", package = "maftools")
relapse.apl = read.maf(maf = relapse.apl)
#Considering only genes which are mutated in at-least in 5 samples in one of the cohort to avoid bias due to genes mutated in single sample.
pt.vs.rt <- mafCompare(m1 = primary.apl, m2 = relapse.apl, m1Name = 'Primary', m2Name = 'Relapse', minMut = 5)
print(pt.vs.rt)
```
forest plots
```{r}

forestPlot(mafCompareRes = pt.vs.rt, pVal = 0.1, color = c('royalblue', 'maroon'), geneFontSize = 0.8)
```
Co-onco plots
```{r}
genes = c("PML", "RARA", "RUNX1", "ARID1B", "FLT3")
coOncoplot(m1 = primary.apl, m2 = relapse.apl, m1Name = 'PrimaryAPL', m2Name = 'RelapseAPL', genes = genes, removeNonMutated = TRUE)
```
Lollipop plot-2
```{r}
lollipopPlot2(m1 = primary.apl, m2 = relapse.apl, gene = "PML", AACol1 = "amino_acid_change", AACol2 = "amino_acid_change", m1_name = "Primary", m2_name = "Relapse")

```
Clinical enrichment analysis
```{r}
fab.ce = clinicalEnrichment(maf = laml.f, clinicalFeature = 'FAB_classification')
fab.ce$groupwise_comparision[p_value < 0.05]
plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05)

```
Drug-Gene Interactions 
```{r}
dgi = drugInteractions(maf = laml.f, fontSize = 0.75)
dnmt3a.dgi = drugInteractions(genes = "DNMT3A", drugs = TRUE)
#Printing selected columns.
dnmt3a.dgi[,.(Gene, interaction_types, drug_name, drug_claim_name)]

```
Oncogenic Signaling Pathways 
```{r}
OncogenicPathways(maf = laml.f)
PlotOncogenicPathways(maf = laml.f, pathways = "RTK-RAS")

```
```{r}
# required libraries for 9.11
library(BSgenome.Hsapiens.UCSC.hg19)
library(NMF)
library(pheatmap)
```
Mutational signatures 
```{r}

laml.tnm = trinucleotideMatrix(maf = laml.f, prefix = 'chr', add = TRUE, ref_genome = "BSgenome.Hsapiens.UCSC.hg19")
plotApobecDiff(tnm = laml.tnm, maf = laml.f, pVal = 0.2)
laml.sign = estimateSignatures(mat = laml.tnm, nTry = 4,  pConstant =0.1)
plotCophenetic(res = laml.sign)
#Compate against original 30 signatures 
laml.og30.cosm = compareSignatures(nmfRes = laml.sign, sig_db = "legacy")
laml.v3.cosm = compareSignatures(nmfRes = laml.sign, sig_db = "SBS")
pheatmap::pheatmap(mat = laml.og30.cosm$cosine_similarities, cluster_rows = FALSE, main = "cosine similarity against validated signatures")
maftools::plotSignatures(nmfRes = laml.sig, title_size = 0.8)
```
Signture enrichment analysis
```{r}
laml.se = signatureEnrichment(maf = laml.f, sig_res = laml.sign)
plotEnrichmentResults(enrich_res = laml.se, pVal = 0.05)
```



